tools/src/wyvern/tools/parsing/coreparser/arch/ASTConnectorTypeDecl.java

```diff
@@ -0,0 +1,107 @@
+import wyvern.target.corewyvernIL.type.NominalType;
+import wyvern.target.corewyvernIL.type.StructuralType;
+import wyvern.tools.errors.ErrorMessage;
+import wyvern.tools.errors.HasLocation;
+import wyvern.tools.errors.ToolError;
+
+public class ASTConnectorTypeDecl extends SimpleNode {
+    private String typeName;
+    private HashMap<String, String> vals = new HashMap<>();
+
+    public ASTConnectorTypeDecl(int id) {
+        super(id);
+    }
+
+    public ASTConnectorTypeDecl(ArchParser p, int id) {
+        super(p, id);
+    }
+
+    public String getTypeName() {
+        return typeName;
+    }
+
+    public void setTypeName(String t) {
+        typeName = t;
+    }
+
+    public String toString() {
+        return super.toString() + " " + typeName;
+    }
+
+    public void collectVals() {
+        for (Node child : this.getChildren()) {
+            if (child instanceof ASTValDecl) {
+                String name = ((ASTValDecl) child).getName();
+                String type = ((ASTValDecl) child).getType();
+                vals.put(name, type);
+            }
+        }
+    }
+
+    public boolean checkModule(InterpreterState state) {
+        Module mod = null;
+        try {
+            mod = state.getResolver().resolveType(typeName + "Properties");
+            for (HasLocation i : ((SeqExpr) mod.getExpression()).getElements()) {
+                IExpr expr = ((VarBinding) i).getExpression();
+                if (expr instanceof New) {
+                    New n = (New) expr;
+                    for (Declaration d : n.getDecls()) {
+                        if (d instanceof TypeDeclaration // contains metadata
+                                && d.getName().equals(typeName + "Properties")) {
+                            StructuralType t = (StructuralType) ((TypeDeclaration) d)
+                                    .getSourceType();
+                            for (DeclType dt : t.getDeclTypes()) {
+                                ValDeclType vdt = (ValDeclType) dt;
+                                String name = vdt.getName();
+                                String type = ((NominalType) vdt.getSourceType())
+                                        .getTypeMember();
+                                if (!vals.get(name).equals(type)) {
+                                    ToolError.reportError(
+                                            ErrorMessage.CONNECTOR_VAL_INCONSISTENCY, this.getLocation(),
+                                            typeName);
+                                }
+                            }
+                            return true;
+                        }
+                    }
+                    ToolError.reportError(ErrorMessage.TYPE_NOT_DEFINED, this.getLocation(),
+                            typeName);
+                }
+            }
+        } catch (ToolError e) {
+            e.printStackTrace();
+            return false;
+        }
```

Comment:
I think the try/catch block should be removed. This function should only return a boolean value indicating whether the check passed or failed. The exception should be thrown in DeclCheckVisitor.